recipes/tinyspline/all/patches/0.5.0-fix-parson.patch

```diff
@@ -0,0 +1,168 @@
+--- CMakeLists.txt
++++ CMakeLists.txt
+@@ -7,6 +7,7 @@ set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+ option(TINYSPLINE_BUILD_EXAMPLES "Build TinySpline examples." ON)
+ option(TINYSPLINE_BUILD_TESTS "Build TinySpline tests." ON)
+ option(TINYSPLINE_BUILD_DOCS "Build TinySpline documentation." ON)
++option(TINYSPLINE_JSON_SUPPORT "Build TinySpline with JSON support." ON)
+ 
+ if(NOT DEFINED TINYSPLINE_OUTPUT_DIRECTORY)
+   set(TINYSPLINE_OUTPUT_DIRECTORY
+--- src/CMakeLists.txt
++++ src/CMakeLists.txt
+@@ -884,7 +884,6 @@ set(TINYSPLINE_CXX_INCLUDE_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
+ # TINYSPLINE_C_SOURCE_FILES
+ list(APPEND TINYSPLINE_C_SOURCE_FILES
+      "${CMAKE_CURRENT_SOURCE_DIR}/tinyspline.c"
+-     "${CMAKE_CURRENT_SOURCE_DIR}/parson.c"
+ )
+ 
+ # TINYSPLINE_CXX_SOURCE_FILES
+@@ -912,6 +911,9 @@ list(JOIN TINYSPLINE_C_INSTALL_CMAKE_CONFIG_DIR "/"
+ 
+ add_library(tinyspline ${TINYSPLINE_C_SOURCE_FILES})
+ target_compile_definitions(tinyspline PUBLIC ${TINYSPLINE_C_DEFINITIONS})
++if (NOT TINYSPLINE_JSON_SUPPORT)
++  target_compile_definitions(tinyspline PRIVATE NO_JSON)
++endif()
+ set_target_properties(
+   tinyspline
+   PROPERTIES OUTPUT_NAME "${TINYSPLINE_C_LIBRARY_OUTPUT_NAME}"
+@@ -1006,6 +1008,9 @@ if(TINYSPLINE_ENABLE_CXX)
+     endif()
+     add_library(${target} ${TINYSPLINE_CXX_SOURCE_FILES})
+     target_compile_definitions(${target} PUBLIC ${TINYSPLINE_CXX_DEFINITIONS})
++    if (NOT TINYSPLINE_JSON_SUPPORT)
++      target_compile_definitions(${target} PRIVATE NO_JSON)
++    endif()
+     set_target_properties(
+       ${target}
+       PROPERTIES OUTPUT_NAME ${name}
+--- src/tinyspline.c
++++ src/tinyspline.c
+@@ -1,6 +1,9 @@
+ #define TINYSPLINE_EXPORT
+ #include "tinyspline.h"
+-#include "parson.h" /* serialization */
++
++#ifndef NO_JSON
++# include "parson.c" /* serialization */
```

Comment:
@SpaceIm here, I'm including `parson.c` instead of `parson.h` therefore it's not needed anymore in `CMakeLists.txt`

Reply:
It's not very elegant and more work for future maintainers if we want to unvendor parson in another PR.